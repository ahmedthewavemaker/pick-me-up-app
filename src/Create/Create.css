

.create{
    display: flex;
    justify-content: center;
}

.build{
    align-content: center;
    align-items: center;
    text-align: left;
}

.submit{
    font-size: 20px;
    background-color: rebeccapurple;
}