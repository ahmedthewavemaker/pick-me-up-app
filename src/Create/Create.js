import React, {useState} from 'react'
import './Create.css'
import GameApiService from '../services/games-service'

function Create(props){

    const [error, setError] = useState(null)

    const handleSubmit=(ev)=>{
       ev.preventDefault()
       setError(null)
       const {name, maxplayers, location, date} = ev.target;

       GameApiService.postGame({
           name: name.value,
           maxplayers: maxplayers.value,
           location: location.value,
           date: date.value
        })

        .then(game => {
             name.value =''
             maxplayers.value=''
             location.value=''
            date.value=''
            props.history.push('/games')

        })

        .catch(res => {
            console.log(res)
            setError( res.error )
          })



    }


    return(
        <div className= 'create'>
            <form onSubmit={handleSubmit} className='build'>
                {error}
                <label className='build' htmlFor='name'> Game name: </label>
                <input type='text' name= 'name' id='name'/>
                <br/>
                <label className='build' htmlFor='maxplayers'> Max Players: </label>
                <input type='number' name= 'maxplayers' id='maxplayers'/>
                <br/>
                <label className='build' htmlFor='location'> Parks/Fields: </label>
                <input type='text' name= 'location' id='location'/>
                <br/>
                <label className='build' htmlFor='date'> Date of Play: </label>
                <input type='text' name= 'date' id='date'/>
                <br/>
                <button className= 'submit'> Post</button>


                
            </form>
        </div>
    )
}

export default Create;