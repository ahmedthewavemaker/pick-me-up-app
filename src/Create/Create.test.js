import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Create from './Create';
import { MemoryRouter } from 'react-router-dom';

describe('Login component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter> <Create />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter> <Create  />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})