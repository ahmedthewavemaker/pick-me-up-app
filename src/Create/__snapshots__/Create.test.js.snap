// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="create"
  >
    <form
      className="build"
      onSubmit={[Function]}
    >
      <label
        className="build"
        htmlFor="name"
      >
         Game name: 
      </label>
      <input
        id="name"
        name="name"
        type="text"
      />
      <br />
      <label
        className="build"
        htmlFor="maxplayers"
      >
         Max Players: 
      </label>
      <input
        id="maxplayers"
        name="maxplayers"
        type="number"
      />
      <br />
      <label
        className="build"
        htmlFor="location"
      >
         Parks/Fields: 
      </label>
      <input
        id="location"
        name="location"
        type="text"
      />
      <br />
      <label
        className="build"
        htmlFor="date"
      >
         Date of Play: 
      </label>
      <input
        id="date"
        name="date"
        type="text"
      />
      <br />
      <button
        className="submit"
      >
         Post
      </button>
    </form>
  </div>,
]
`;
